test/performance/io/format_sam_benchmark.cpp

```diff
@@ -63,15 +63,15 @@ static std::string create_sam_file_string(size_t const n_queries)
         {
             auto query = seqan3::test::generate_sequence<seqan3::dna4>(read_size, length_variance, seed + i);
             auto qualities = seqan3::test::generate_sequence<seqan3::phred42>(read_size, length_variance, seed + i);
-            auto align_result = *(seqan3::align_pairwise(std::tie(query, reference), config).begin());
+            auto align_result = *(seqan3::align_pairwise(std::tie(reference, query), config).begin());
             std::string const current_query_id = query_prefix + std::to_string(i);
 
             sam_out.emplace_back(query,                                   // field::seq
                                  current_query_id,                        // field::id
-                                 align_result.sequence1_begin_position(), // field::offset
+                                 align_result.sequence2_begin_position(), // field::offset
                                  reference,                               // field::ref_seq
                                  reference_id,                            // field::ref_id
-                                 align_result.sequence2_begin_position(), // field::ref_offset
+                                 align_result.sequence1_begin_position(), // field::ref_offset
```

Comment:
Ah now I see. So the actual fix is that the leading and trailing gaps were configured wrongly for `(query, reference)` but are now correct with `(reference, query)`. The resulting (maybe weird) alignment caused the error when writing SAM.
Do I understand this correctly? Because then this just circumvents the error and does not fix it?

Reply:
Well, I have to find the line of code, but the problem is that inferring the CIGAR from the alignment in the SAM output implicitly requires the first sequence to be the reference

https://github.com/seqan/seqan3/blob/master/include/seqan3/io/alignment_file/format_sam.hpp#L840-L853

`<1>` is accessed, and IIRC, `<1>` needs to be the query.

Reply:
Thanks for pointing out the code. 

Can you add the following assert before the size is subtracted from `off_end`:

```cpp
assert(off_end >= std::ranges::size(get<1>(align)); // Might happen if get<1>(align) doesn't correspond to the reference.
```

I even think that this might be a case were we should throw but another if branch would of course impact the performance. I would add the assert for now and re-evaluate this once we have a stable IO interface and start working on the performance.